packages/react-storage/src/components/StorageBrowser/components/PaginateButton.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { CLASS_BASE } from '../views/constants';
+import {
+  ButtonElement,
+  IconElement,
+  InputElementProps,
+} from '../context/elements';
+
+const BLOCK_NAME = `${CLASS_BASE}__paginate`;
+
+interface PaginateButtonProps extends InputElementProps {
+  disabled: boolean;
+  onClick: () => void;
+  type: 'previous' | 'next';
+}
+export function PaginateButton({
+  disabled,
+  onClick,
+  type,
+}: PaginateButtonProps): React.JSX.Element {
+  return (
+    <ButtonElement
+      aria-label={`Go to ${type} page`}
+      className={`${BLOCK_NAME}__button-${type}`}
```

Comment:
Regardless of what we end up doing as the _block_, we should flatten the elements
```ts
      className={`${CLASS_BASE}__paginate-button-${type}`}
```

I am not sure but I don't think variant automatically end up as a modifier of the class, right? If it does, maybe we can just drop `-${type}` here. Otherwise, I think it'd make sense to treat this section of the className as modifier... i.e.
```ts
      className={`${CLASS_BASE}__paginate-button--paginate-${type}`}
```

Reply:
Updated. I don't believe the variant is added automatically, so I added it using a reusable const